### DISCUSSION TITLE
Support for pydantic-settings

### DISCUSSION THREAD

**Discussion by siddhesh1981** [BeanieODM/beanie#617] [Feature Request]
Currently Beanie supports "pydantic>=1.10.0,<2.0.0". Pydantic-settings which allows settings management using pydantic and has a good support for loading environment variables, only supports pydantic>=2.0.1. 

It will be great if Beanie starts supporting pydantic=2.0.1 or greater, so that the new improved way of getting environment variables  can be incorporated in fastapi projects that need beanie, using pydantic-settings.

Reference:
https://fastapi.tiangolo.com/ru/advanced/settings/
https://pypi.org/project/pydantic-settings/

Thanks!

**Comment by roman-right** [comment]
Hi!
I'm working on supporting Pydantic V2 ATM. For now, you can use BaseSettings from the v1. 
V2 support beta will be published next week.

**Reply by siddhesh1981** [reply]
Thanks! That will be helpful.
